src/components/KTableView/KTableView.vue

```diff
@@ -880,6 +1026,80 @@ watch(hasColumnVisibilityMenu, (newVal) => {
     columnVisibility.value = props.tablePreferences.columnVisibility || {}
   }
 }, { immediate: true })
+
+const bulkActionsAll = ref<boolean>(false)
+
+const isBulkActionsIndeterminate = computed((): boolean => {
+  // ignore thee disabled rows
+  const selectableRows = tableData.value.filter((row) => getRowBulkActionEnabled(row))
+
+  // it is indeterminate if there are selected and unselected rows
+  return !!selectableRows.filter((row) => row.selected).length && !!selectableRows.filter((row) => !row.selected).length
+})
+
+const handleIndeterminateChange = (value: boolean) => {
+  if (value) {
+    // select all selectable rows
+    tableData.value = [...tableData.value].map((row) => ({ ...row, selected: getRowBulkActionEnabled(row) }))
+  } else {
+    // unselect and reset all
+    tableData.value = [...props.data].map((row) => ({ ...row, selected: false }))
+  }
+}
+
+watch(tableData, (newVal) => {
+  /** update the bulkActionsAll value */
+
+  const selectableRows = newVal.filter((row) => getRowBulkActionEnabled(row))
+
+  // all are selected
+  if (selectableRows.filter((row) => row.selected).length === selectableRows.length) {
+    bulkActionsAll.value = true
+  // all are unselected
+  } else if (selectableRows.filter((row) => !row.selected).length === selectableRows.length) {
+    bulkActionsAll.value = false
+  // some are selected
+  } else {
+    bulkActionsAll.value = false
+  }
+
+  /** update the selected rows */
+
+  const newSelectedRows = newVal.filter((row) => row.selected).map(({ selected, ...rest }) => rest)
+
+  const oldSelectedRows: TableViewData = []
+  bulkActionsSelectedRows.value.forEach((selectedRow) => {
+    const row = props.data.find((row) => JSON.stringify(row) === JSON.stringify(selectedRow))
+
+    if (!row) {
+      oldSelectedRows.push(selectedRow)
+    }
+  })
+
+  bulkActionsSelectedRows.value = [...oldSelectedRows, ...newSelectedRows]
+}, { deep: true })
+
+/**
+ * Watch for changes in the data prop and update the tableData
+ * We want to display the selected rows from the previous data prop value
+ */
+watch(() => props.data, (newVal) => {
+  tableData.value = []
+
+  newVal.forEach((row) => {
+    const selectedRow = bulkActionsSelectedRows.value.find((selectedRow) => JSON.stringify(selectedRow) === JSON.stringify(row))
+
+    if (selectedRow) {
+      tableData.value.push({ ...row, selected: true })
+    } else {
+      tableData.value.push({ ...row, selected: false })
+    }
+  })
+}, { deep: true })
+
+watch(bulkActionsSelectedRows, (newVal) => {
+  emit('bulk-actions-select', newVal)
+}, { deep: true })
```

Comment:
question: does this need to be a deep watcher? if the array count changes, the watcher would be triggered; I don't think you need to watch the actual data within each row?

Reply:
You're right, fixed